pkg/controller/pv_controller_sync.go

```diff
@@ -0,0 +1,219 @@
+		// If it is not required to send event
+		return nil
+	}
+
+	klog.Infof("Got PV %s to send volume events", pvObj.Name)
+	eventSender, err := pController.getEventSender(pvObj)
+	if err != nil {
+		return err
+	}
+	if eventSender == nil {
+		return nil
+	}
+
+	// Send create event information
+	err = pController.sendCreateEvent(eventSender, pvObj)
+	if err != nil {
+		return err
+	}
+
+	// Send delete event information
+	err = pController.sendDeleteEvent(eventSender, pvObj)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// sendCreateEvent will push create volume event to configured server
+// NOTE: If event is already sent then sendCreateEvent will return nil
+func (pController *PVEventController) sendCreateEvent(
+	eventSender collectorinterface.EventsSender,
+	pvObj *corev1.PersistentVolume) error {
+
+	// Send create information
+	if !isCreateVolumeEventSent(pvObj) {
+		// Get create event related data
+		data, err := eventSender.CollectCreateEvents()
+		if err != nil {
+			return errors.Wrapf(err, "failed to get create event data of volume %s", pvObj.Name)
+		}
+
+		// Send create event data
+		err = eventSender.Send(data)
+		if err != nil {
+			return errors.Wrapf(err, "failed to send create event data of volume %s to server", pvObj.Name)
+		}
+
+		_, err = eventSender.AnnotateCreateEvent(pvObj)
+		if err != nil {
+			return err
+		}
+		klog.Infof("Successfully sent create volume %s event to server", pvObj.Name)
+	}
+	return nil
+}
+
+// sendDeleteEvent will push delete volume events to configured server
+// NOTE: If event is already sent then sendDeleteEvent will return nil
+func (pController *PVEventController) sendDeleteEvent(
+	eventSender collectorinterface.EventsSender,
+	pvObj *corev1.PersistentVolume) error {
+	if pvObj.DeletionTimestamp != nil {
+		if !isDeleteVolumeEventSent(pvObj) {
+			// Get delete event related data
+			data, err := eventSender.CollectDeleteEvents()
+			if err != nil {
+				return errors.Wrapf(err, "failed to get delete event data of volume %s", pvObj.Name)
+			}
+
+			// Send delete event data
+			err = eventSender.Send(data)
+			if err != nil {
+				return errors.Wrapf(err, "failed to send delete event data of volume %s to server", pvObj.Name)
+			}
+
+			_, err = eventSender.AnnotateDeleteEvent(pvObj)
+			if err != nil {
+				return errors.Wrapf(err, "failed to annotate volume %s with delete event information", pvObj.Name)
+			}
+			klog.Infof("Successfully sent delete volume %s event to server", pvObj.Name)
+		}
+
+		err := eventSender.RemoveEventFinalizer()
+		if err != nil {
+			return errors.Wrapf(err, "failed to remove finalizers on volume %s", pvObj.Name)
+		}
+	}
+	return nil
+}
+
+func (pController *PVEventController) getEventSender(pvObj *corev1.PersistentVolume) (collectorinterface.EventsSender, error) {
+	// Add more types based on underlying volume type
+	if value, ok := pvObj.Labels[nfspv.OpenEBSCASLabelKey]; ok && value == nfspv.OpenEBSNFSCASLabelValue {
```

Comment:
nit: let's use switch case here

Reply:
Updated the code... Which can also accumulate CSI based PVs